compiler/rustc_codegen_llvm/src/coverageinfo/ffi.rs

```diff
@@ -200,22 +303,52 @@ impl CounterMappingRegion {
     pub(crate) fn branch_region(
         counter: Counter,
         false_counter: Counter,
+        condition_info: Option<ConditionInfo>,
```

Comment:
I think this constructor function should only create plain branch regions. MC/DC branch regions should have their own constructor.